multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakEnvironmentInitalizationFilter.java

```diff
@@ -92,11 +92,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         }
         String email = claims.get("email", String.class);
         if (isNullOrEmpty(email)) {
-          sendError(
-              response,
-              400,
-              "Unable to authenticate user because email address is not set in keycloak profile");
-          return;
+          email = username + "@" + username;
```

Comment:
Oh. Personally I don't like that idea much. Maybe we can require email via [rules](https://auth0.com/docs/rules/current) try to find another solution.

Reply:
I agree that we have to find another solution. Your remark was expected and this is one of the points we have to discuss to find the right implementation.

However rules are an Auth0-only thing afaict, and we cannot rely on production-specific things that are not in the specification.

I assume that the best thing would be remove the requirement to have the email during the user creation in the Authentication Filter, and in the case we don't have it at this time, allow retrieving it when contacting the `userinfo` endpoint during profile creation / update.

wdyt ?